src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -684,6 +684,12 @@ private bool OperationUsesPreviewFeatures(OperationAnalysisContext context, Conc
                     symbol = eventAssignment.Adds ? eventSymbol.AddMethod : eventSymbol.RemoveMethod;
                 }
 
+                if (symbol == null)
```

Comment:
On line 678 above, propertySymbol.SetMethod could be null. I'm just being defensive and adding the null check here.

Reply:
I would not expect `SetMethod` to be null when `usageInfo.IsWrittenTo()` but `usageInfo == ValueUsageInfo.ReadWrite` might be, do not know for sure. 

I think it is better to set the symbol with `SetMethod` only if the getter was not available, i.e for lines 671-679:
```cs
        if (usageInfo.IsReadFrom())
        {
            symbol = propertySymbol.GetMethod;
        }
        else if (usageInfo.IsWrittenTo() || usageInfo == ValueUsageInfo.ReadWrite)
        {
            symbol = propertySymbol.SetMethod;
        }
```
Because property is expected to have a getter or setter (or both of course) no need to check for null after this

Reply:
And I expect when there is a getter we should check/analyzer the getter

Reply:
> And I expect when there is a getter we should check/analyzer the getter

Ah i misunderstood, getter was not overwritten, please ignore the above comments

I wonder if we should continue the analysis with the original `symbol` (set the symbol with only if setter or getter is not null), which might be not needed as the property cannot be set from outside